download_from_csv.py

```diff
@@ -1,41 +1,150 @@
+    csv_data = str(csv_data)
+    try:
+        f = open('playlist.csv', 'w+')
+        f.write(csv_data)
+        f.close()
+    except Exception as e:
+        raise Exception("Couldn't write csv file to disk")
+    return 'playlist.csv'
+
+# Command line parser arguments
 parser = argparse.ArgumentParser()
-parser.add_argument('-f', '--folder', help="keep the files in the folder specified")
-parser.add_argument('-c', '--create', help="try to create folder if doesn't exist",
-                    action="store_true")
-parser.add_argument('--skip', help="number of songs to skip from the start of csv",
-                    type=int)
-parser.add_argument('csv', help="input csv file")
+parser.add_argument('-f', '--folder', default=os.getcwd(),
+                    help="keep the files in the folder specified")
+parser.add_argument('-c', '--create', action="store_true",
+                    help="try to create folder if doesn't exist")
+parser.add_argument('--skip', type=int,
+                    help="number of songs to skip from the start of csv")
+parser.add_argument('-s', '--search', nargs='*',
+                    help="source for songs, default it youtube")
+parser.add_argument('-p', '--playlist',
+                    help="spotify playlist URI(requires username, auth token)")
+parser.add_argument('-u', '--username', help="Enter your username")
+parser.add_argument('-t', '--token',
+                    help="Enter authorization token, get it here: https://d" +
+                    "evloper.spotify.com/web-api/console/get-playlist-tracks/")
+parser.add_argument('-i', '--csv', help="input a csv file")
 args = parser.parse_args()
 
-folder = ''
-if os.path.isfile(args.csv):
-    csvfile = args.csv
+csvfile = None
+if args.csv:
+    if os.path.isfile(args.csv):
+        csvfile = args.csv
+    else:
+        print 'No such csv file. Aborting..'
+        exit()
+elif args.playlist:
+    username = None
+    token = None
+    if args.username:
+        username = args.username
+    else:
+        username = raw_input("\033[93mEnter username to access api: \033[0m")
+    if args.token:
+        token = args.token
+    else:
+        token = raw_input(
+                          "if you don't have one, get it here: \033[94m" +
+                          "https://developer.spotify.com/web-api/console/ge" +
+                          "t-playlist-tracks/\n\033[93mEnter fresh token to" +
+                          " access api: \033[0m"
+                        )
+    csvfile = generate_csv(args.playlist, username, token)
 else:
-    print 'No such csv file. Aborting..'
-    exit()
+    parser.error("No action requested, add -p or -i. Or add -h for help")
+
 
+folder = ''  # will be set to current if it's not given in arguments
 if args.folder:
+    args.folder = os.path.relpath(args.folder)
     if os.path.isdir(args.folder):
-        folder = os.path.abspath(args.folder)
+        folder = os.path.relpath(args.folder)
+        folder = args.folder
```

Comment:
What are you doing here?
Isn't this statement overwriting the value of the previous statement? If yes, which one of these is redundant?